static/app/components/modals/sudoModal.tsx

```diff
@@ -42,36 +42,63 @@ type State = {
   busy: boolean;
   error: boolean;
   errorType: string;
+  showAccessForms: boolean;
   superuserAccessCategory: string;
   superuserReason: string;
 };
 
 class SudoModal extends Component<Props, State> {
   state: State = {
+    authenticators: [],
+    busy: false,
     error: false,
     errorType: '',
-    busy: false,
+    showAccessForms: true,
     superuserAccessCategory: '',
     superuserReason: '',
-    authenticators: [],
   };
 
   componentDidMount() {
     this.getAuthenticators();
   }
 
-  handleSubmit = async () => {
-    const {api, isSuperuser} = this.props;
-    const data = {
-      isSuperuserModal: isSuperuser,
+  handleSubmitCOPS = () => {
+    this.setState({
       superuserAccessCategory: 'cops_csm',
       superuserReason: 'COPS and CSM use',
-    };
-    try {
-      await api.requestPromise('/auth/', {method: 'PUT', data});
-      this.handleSuccess();
-    } catch (err) {
-      this.handleError(err);
+    });
+  };
+
+  handleSubmit = async data => {
+    const {api, isSuperuser} = this.props;
+    const superuserAccessCategory =
+      this.state.superuserAccessCategory === null
+        ? data.superuserAccessCategory
+        : this.state.superuserAccessCategory;
+
+    const superuserReason =
+      this.state.superuserReason === null
+        ? data.superuserReason
+        : this.state.superuserReason;
+
+    if (!this.state.authenticators.length) {
```

Comment:
we can deconstruct the state at the top 
```
const { superuserAccessCategory, superuserReason, authenticators }  = this.state
```

and we can probably do this check at the top too so we return early